ext/webserial/lib.deno_webserial.d.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/// <reference no-default-lib="true" />
+/// <reference lib="esnext" />
+
+declare class Serial {
+  getPorts(): Promise<SerialPort[]>;
+  requestPort(options?: SerialPortRequestOptions): Promise<SerialPort>;
+}
+
+declare class SerialPort {
+  readonly readable: ReadableStream<Uint8Array> | null;
+  readonly writable: WritableStream<Uint8Array> | null;
+
+  getInfo(): SerialPortInfo;
+
+  open(options: SerialOptions): Promise<void>;
+  setSignals(signals?: SerialOutputSignals): Promise<void>;
+  getSignals(): Promise<SerialInputSignals>;
+  close(): Promise<void>;
+}
+
+declare interface SerialPortRequestOptions {
+  filters?: SerialPortFilter[];
+}
+
+declare interface SerialPortFilter {
+  usbVendorId?: number;
+  usbProductId?: number;
+}
+
+declare interface SerialPortInfo {
```

Comment:
Could port name be added to `SerialPortInfo`? Given even the `--allow-serial` permission works based on serial port names. And without having access to port's name you can't *easily* connect to a specific port by name.